Scripts - PCB/NofittedNoPaste/NofittedNoPaste.pas

```diff
@@ -0,0 +1,89 @@
+
+Procedure StartScript ();
+var
+  Board       : IPCB_Board;
+  ComponentPCB: IPCB_Component;
+  boolAss     : boolean;
+  boolused    : boolean;
+  NoFitye     : boolean;
+  RulePaste   : IPCB_PasteMaskExpansionRule;
+
+  filter      : string;
+
+  Workspace     : IWorkspace;
+  PCBProject    : IProject;
+  VariantNum    : Integer;
+  Variant       : IProjectVariant;
+  FlatHierarchy   : IDocument;
+  Component       : IComponent;
+  ComponentNum    : Integer;
+  CompVariation   : IComponentVariation;
+
+  fitted : boolean;
+
+Begin
+  NoFitye := true;
+  boolAss := False;
+  boolused := False;
+  fitted := True;
+  Workspace   := GetWorkspace;
+  PCBProject  := Workspace.DM_FocusedProject;
+
+  filter := '(ObjectKind = ''Pad'') AND (';
+
+  Board := PCBServer.GetCurrentPCBBoard;
+  If Board = nil then Begin ShowError('Open board!'); Exit; End;
+
+
+
+  If (PcbProject = nil) then
+   begin
+      ShowMessage('Current Project is not a PCB Project');
+      exit;
+   end;
+
+   If (AnsiUpperCase(ExtractFileExt(PCBProject.DM_ProjectFileName)) <> '.PRJPCB') then
+   begin
+      ShowMessage('Current Project is not a PCB Project');
+      exit;
+   end;
+
+   Variant := PCBProject.DM_CurrentProjectVariant;
+
+   FlatHierarchy := PCBProject.DM_DocumentFlattened;
+
+   if Variant = NIL then
+   begin
+      ShowMessage('Select a board Variant.');
+      exit;
+   end;
+
+   for ComponentNum := 0 to FlatHierarchy.DM_ComponentCount - 1 do
+   begin
+      Component := FlatHierarchy.DM_Components(ComponentNum);
+      CompVariation := Variant.DM_FindComponentVariationByUniqueId(Component.DM_UniqueId);
+      if (CompVariation <> NIL) then
+      if (CompVariation.DM_VariationKind = eVariation_NotFitted) then
+      begin
+           filter := filter + '(Component = ''' + Component.DM_PhysicalDesignator + ''') OR ';
+           NoFitye := false;
+      end;
+   end;
+  filter := copy(filter,1,Length(filter) - 4) + ')';
+
+  RulePaste := PCBServer.PCBRuleFactory(eRule_PasteMaskExpansion);
+  RulePaste.Expansion := MMsToCoord(-100);
```

Comment:
Would be good to have a number that is **really** impossible to obtain in practice (-10 000?).

Reply:
I have never made pads more than 100mm. It seemed to me enough. But more is possible.

Reply:
I tested -10000. The result of converting such a large number is incorrect: 909.217mm

Reply:
![10000](https://user-images.githubusercontent.com/10257640/84024323-e4bc3080-a9b3-11ea-9a55-265ffaa82304.PNG)

Reply:
Empirically, set -5000. Tested.

Reply:
Additional tests showed that -5000 also failed.  -2539.975mm maximum.